components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1074,79 +1005,39 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
-                } catch (error) {
-                    this.trackEvent(ctx, {
-                        event: "snapshot_access_denied",
-                        properties: { snapshot_id: context.snapshotId, error: String(error) },
-                    }).catch((err) => log.error("cannot track event", err));
-                    if (UnauthorizedError.is(error)) {
-                        throw error;
-                    }
-                    throw new ApplicationError(
-                        ErrorCodes.PERMISSION_DENIED,
-                        `Snapshot URLs require read access to the underlying repository. Please request access from the repository owner.`,
-                    );
-                }
-                this.trackEvent(ctx, {
-                    event: "snapshot_access_granted",
-                    properties: { snapshot_id: context.snapshotId },
-                }).catch((err) => log.error("cannot track event", err));
-            }
-
-            // if we're forced to use the default config, mark the context as such
-            if (!!options.forceDefaultConfig) {
-                context = WithDefaultConfig.mark(context);
-            }
-
-            // if this is an explicit prebuild, check if the user wants to install an app.
-            if (
-                StartPrebuildContext.is(context) &&
-                CommitContext.is(context.actual) &&
-                context.actual.repository.cloneUrl
-            ) {
-                const cloneUrl = context.actual.repository.cloneUrl;
-                const host = new URL(cloneUrl).hostname;
-                const hostContext = this.hostContextProvider.get(host);
-                const services = hostContext && hostContext.services;
-                if (!hostContext || !services) {
-                    console.error("Unknown host: " + host);
-                } else {
-                    // on purpose to not await on that installation process, because it‘s not required of workspace start
-                    // See https://github.com/gitpod-io/gitpod/pull/6420#issuecomment-953499632 for more detail
-                    (async () => {
-                        if (await services.repositoryService.canInstallAutomatedPrebuilds(user, cloneUrl)) {
-                            console.log("Installing automated prebuilds for " + cloneUrl);
-                            await services.repositoryService.installAutomatedPrebuilds(user, cloneUrl);
+            const { context, project } = await this.contextService.parseContext(
+                user,
+                contextUrl,
+                {
+                    projectId: options.projectId,
+                    organizationId: options.organizationId,
+                },
+                (context) => {
+                    // if this is an explicit prebuild, check if the user wants to install an app.
+                    if (
```

Comment:
cannot we just call it after calling parseContext? we already get returned context anyway?